Tools/LambdaTestTool/src/Amazon.Lambda.TestTool/TestToolStartup.cs

```diff
@@ -0,0 +1,326 @@
+                {
+                    lambdaAssemblyDirectory = Path.Combine(lambdaAssemblyDirectory, $"../../tests/LambdaFunctions/netcore31/S3EventFunction/bin/Debug/{targetFramework}");
+                }
+                // If running in the project directory select the build directory so the deps.json file can be found.
+                else if (Utils.IsProjectDirectory(lambdaAssemblyDirectory))
+                {
+                    lambdaAssemblyDirectory = Path.Combine(lambdaAssemblyDirectory, $"bin/Debug/{targetFramework}");
+                }
+
+                localLambdaOptions.LambdaRuntime = LocalLambdaRuntime.Initialize(lambdaAssemblyDirectory);
+                runConfiguration.OutputWriter.WriteLine($"Loaded local Lambda runtime from project output {lambdaAssemblyDirectory}");
+
+                if (commandOptions.NoUI)
+                {
+                    ExecuteWithNoUi(localLambdaOptions, commandOptions, lambdaAssemblyDirectory, runConfiguration);
+                }
+                else
+                {
+                    // Look for aws-lambda-tools-defaults.json or other config files.
+                    localLambdaOptions.LambdaConfigFiles = Utils.SearchForConfigFiles(lambdaAssemblyDirectory);
+
+                    // Start the test tool web server.
+                    uiStartup(localLambdaOptions, !commandOptions.NoLaunchWindow);
+                }
+            }
+            catch (CommandLineParseException e)
+            {
+                runConfiguration.OutputWriter.WriteLine($"Invalid command line arguments: {e.Message}");
+                runConfiguration.OutputWriter.WriteLine("Use the --help option to learn about the possible command line arguments");
+                if (runConfiguration.Mode == RunConfiguration.RunMode.Normal)
+                {
+                    if (Debugger.IsAttached)
+                    {
+                        Console.WriteLine("Press any key to exit");
+                        Console.ReadKey();
+                    }
+                    System.Environment.Exit(-1);
+                }
+            }
+            catch (Exception e)
+            {
+                runConfiguration.OutputWriter.WriteLine($"Unknown error occurred causing process exit: {e.Message}");
+                runConfiguration.OutputWriter.WriteLine(e.StackTrace);
+                if (runConfiguration.Mode == RunConfiguration.RunMode.Normal)
+                {
+                    if (Debugger.IsAttached)
+                    {
+                        Console.WriteLine("Press any key to exit");
+                        Console.ReadKey();
+                    }
+                    System.Environment.Exit(-2);
+                }
+            }
+        }
+        
+        public static void ExecuteWithNoUi(LocalLambdaOptions localLambdaOptions, CommandLineOptions commandOptions, string lambdaAssemblyDirectory, RunConfiguration runConfiguration)
```

Comment:
Suggest splitting this function into smaller functions

Reply:
Yeah that method got a bit carried away. Its broken up now.